src/provider/provider_os_memory.c

```diff
@@ -412,7 +413,7 @@ static umf_result_t os_alloc(void *provider, size_t size, size_t alignment,
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
-    int protection = os_provider->protection;
+    int protection = (int)os_provider->protection;
```

Comment:
Perhaps the `os_provider->protection` should be declared an int? If that's what mmap expects, this would be reasonable.